include/ddc/chunk_span.hpp

```diff
@@ -50,22 +49,14 @@ class ChunkSpan<ElementType, DiscreteDomain<DDims...>, LayoutStridedPolicy, Memo
             "ChunkSpan only supports layout_left, layout_right or layout_stride");
 
 protected:
-    using base_type = ChunkCommon<ElementType, DiscreteDomain<DDims...>, LayoutStridedPolicy>;
-
-    /// the raw mdspan underlying this, with the same indexing (0 might no be dereferenceable)
-    using typename base_type::internal_mdspan_type;
+    using base_type = ChunkCommon<ElementType, SupportType, LayoutStridedPolicy>;
 
 public:
     /// type of a span of this full chunk
-    using span_type
-            = ChunkSpan<ElementType, DiscreteDomain<DDims...>, LayoutStridedPolicy, MemorySpace>;
+    using span_type = ChunkSpan<ElementType, SupportType, LayoutStridedPolicy, MemorySpace>;
 
     /// type of a view of this full chunk
-    using view_type = ChunkSpan<
-            ElementType const,
-            DiscreteDomain<DDims...>,
-            LayoutStridedPolicy,
-            MemorySpace>;
+    using view_type = ChunkSpan<ElementType const, SupportType, LayoutStridedPolicy, MemorySpace>;
 
     using discrete_domain_type = typename base_type::discrete_domain_type;
```

Comment:
In this version of chunk_span.hpp, the only place where discrete_domain_type is used, is to define discrete_element_type below. Moreover, discrete_element_type is not used anymore. Thus both discrete_domain_type and discrete_element_type can be removed.